src/Symfony/Component/Translation/PseudoLocalizationTranslator.php

```diff
@@ -0,0 +1,273 @@
+                'L' => 'Ļ',
+                'M' => 'Ṁ',
+                'N' => 'Ñ',
+                'O' => 'Ö',
+                'P' => 'Þ',
+                'Q' => 'Ǫ',
+                'R' => 'Ŕ',
+                'S' => 'Š',
+                'T' => 'Ţ',
+                'U' => 'Û',
+                'V' => 'Ṽ',
+                'W' => 'Ŵ',
+                'X' => 'Ẋ',
+                'Y' => 'Ý',
+                'Z' => 'Ž',
+                '[' => '⁅',
+                '\\' => '∖',
+                ']' => '⁆',
+                '^' => '˄',
+                '_' => '‿',
+                '`' => '‵',
+                'a' => 'å',
+                'b' => 'ƀ',
+                'c' => 'ç',
+                'd' => 'ð',
+                'e' => 'é',
+                'f' => 'ƒ',
+                'g' => 'ĝ',
+                'h' => 'ĥ',
+                'i' => 'î',
+                'j' => 'ĵ',
+                'k' => 'ķ',
+                'l' => 'ļ',
+                'm' => 'ɱ',
+                'n' => 'ñ',
+                'o' => 'ö',
+                'p' => 'þ',
+                'q' => 'ǫ',
+                'r' => 'ŕ',
+                's' => 'š',
+                't' => 'ţ',
+                'u' => 'û',
+                'v' => 'ṽ',
+                'w' => 'ŵ',
+                'x' => 'ẋ',
+                'y' => 'ý',
+                'z' => 'ž',
+                '{' => '(',
+                '|' => '¦',
+                '}' => ')',
+                '~' => '˞', ]) : $text;
+        }
+
+        if (1.0 < $this->expansionFactor) {
+            $textVisibleLength = false === ($encoding = mb_detect_encoding($visibleText, null, true)) ? \strlen($visibleText) : mb_strlen($visibleText, $encoding);
+            $transMissingLength = (int) (ceil($textVisibleLength * $this->expansionFactor)) - $textVisibleLength;
+            if ($this->brackets) {
+                $transMissingLength -= 2;
+            }
+
+            if (0 < $transMissingLength) {
+                $transPrefixLength = (int) (floor($transMissingLength / 2));
+                $transPrefix = '';
+                for ($i = 0; $i < $transPrefixLength; ++$i) {
+                    $transPrefix .= 0 === $i % 2 ? '~' : ' ';
+                }
+
+                $transSuffixLength = $transMissingLength - $transPrefixLength;
+                $transSuffix = '';
+                for ($i = 0; $i < $transSuffixLength; ++$i) {
+                    $transSuffix .= 0 === $i % 2 ? ' ' : '~';
+                }
+
+                $trans = $transPrefix.$trans.$transSuffix;
+            }
+        }
+
+        if ($this->brackets) {
+            $trans = '['.$trans.']';
+        }
+
+        return $trans;
+    }
+
+    private function getParts(string $originalTrans): array
+    {
+        if (!$this->parseHTML) {
+            return [[true, true, $originalTrans]];
+        }
+
+        $crawler = new Crawler();
+        $crawler->addHtmlContent('<html><body><trans>'.$originalTrans.'</trans></body></html>');
```

Comment:
Why using the Crawler class for that ? You can achieve the same almost as easily with DOMDocument directly (and the support of HTML5 is not an argument, as you don't load it as HTML 5).

Reply:
Thanks, I need to look into it. Reusing the DomCrawler component just felt easier.

Reply:
I removed the DomCrawler component dependency to parse the HTML.